services/src/main/java/org/keycloak/exportimport/ExportImportManager.java

```diff
@@ -95,6 +95,7 @@ public boolean isImportMasterIncludedAtStartup(String dir) {
         if (dir == null) {
             throw new IllegalStateException("Import not enabled");
         }
+        ExportImportConfig.setReplacePlaceholders(true);
```

Comment:
This looks like an odd place to set configuration. Can't this be moved somewhere else (maybe the `ExportImportManager`)?

Reply:
Another alternative is to default to always replacing placeholders. I think we didn't do that before just to limit the scope of the changes, but as long as the core folks agree then we can remove both of these calls.

Reply:
> Do you mean you want to get rid of the ExportImportConfig class? That has the responsibility currently of handling all of these system properties.

No, I meant that I don't think we should be calling `ExportImportConfig.setReplacePlaceholders(true);` from `isImportMasterIncludedAtStartup` which is essentially just a getter/checker, it should not modify the configuration IMHO.

> Another alternative is to default to always replacing placeholders. I think we didn't do that before just to limit the scope of the changes, but as long as the core folks agree then we can remove both of these calls.

That might be a bit out of scope for now.

Reply:
> No, I meant that I don't think we should be calling ExportImportConfig.setReplacePlaceholders(true); from isImportMasterIncludedAtStartup which is essentially just a getter/checker, it should not modify the configuration IMHO.

No, it's not great. It's similar to the issues raised in the TODO about undoing the static state changes. We can move it up, but it's not great in KeycloakApplication either.

Reply:
TBH, it makes more sense to me if it was in `KeycloakApplication` than in `isImportMasterIncludedAtStartup`. Feels cleaner.

Reply:
Let's just resolve this in a follow-up: #33596 - ideally we should be encapsulating as much as possible the usage of static ExportImportConfig logic.

Reply:
Ok, let's do it as a follow-up.